webapp/src/components/ListItem/ListItem.vue

```diff
@@ -22,12 +22,15 @@ export default {
   name: 'ListItem',
   props: ['item'],
   methods: {
-    upvoteItem: function() {
+    upvoteItem() {
       this.item.votes++;
```

Comment:
Do not mutate props directly within the component. Mutate in the parent component and let the child component re-render.

Reply:
See here: https://vuejs.org/v2/guide/components-props.html#One-Way-Data-Flow
and here: https://vuejs.org/v2/guide/migration.html#Prop-Mutation-deprecated

Reply:
You can solve it by creating new object on-the-fly:
```js
this.$emit('update-item', { ...this.item, votes: (this.item.votes + 1) })
```
Or initialize separate reactive state:
```js
data() {
  // you have access to your properties in data()
  this.internalItem = { ...this.item }
}
```